planner/core/exhaust_physical_plans.go

```diff
@@ -1993,61 +1992,16 @@ func (p *LogicalJoin) satisfyIndexJoinHint(join PhysicalPlan) bool {
 	case *PhysicalIndexMergeJoin:
 		innerIdx = ij.getInnerChildIdx()
 		joinMethod = indexMergeJoin
-	default:
-		return false
```

Comment:
Maybe add ```panic("unreachable")```

Reply:
I add a new returned value here to indicate whether this plan is a IndexJoin.